pkg/cmd/cli/cmd/gen-dockerfile.go

```diff
@@ -0,0 +1,57 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/source-to-image/pkg/api"
+	"github.com/openshift/source-to-image/pkg/build/strategies/dockerfile"
+	"github.com/openshift/source-to-image/pkg/util/fs"
+)
+
+func NewCmdGenDockerfile(cfg *api.Config) *cobra.Command {
+
+	genDockerfileCmd := &cobra.Command{
+		Use:   "gen-dockerfile <image> [<dockerfile>]",
+		Short: "Generate a Dockerfile based on the provided builder image",
+		Example: `
+# Generate a Dockerfile from a remote builder image:
+$ s2i gen-dockerfile docker://quay.io/redhat-developer/app-binding-operator
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
@isutton Is there a reason to use `RunE:` instead of `Run:`?

```
// Run: Typically the actual work function. Most commands will only implement this
Run func(cmd *Command, args []string)
// RunE: Run but returns an error
RunE func(cmd *Command, args []string) error
```

I can see other s2i top level commands like `build`, `rebuild`, `completion`, `create` use `Run:`

Reply:
To propagate the error to the caller, if any.

Now looking closely at the referred commands, their implementations differ regarding surfacing the errors to the user-just highlighting this as a reference to the intent of using `RunE`. 

Perhaps this should be changed in another work unit, where we could make use of the existing `errors` pkg from source-to-image, where it implements a concrete `Error` type and has supporting functions to treat and display an error message in all commands.